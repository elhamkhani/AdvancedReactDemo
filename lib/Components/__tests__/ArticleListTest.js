import  React from 'react';
import ArticleList from '../ArticleList';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


describe('ArticleList', () => {
  Enzyme.configure({adapter: new Adapter()});

  const testProps = {
    articles:{
      a: {id:'a', authorId:''},
      b: {id:'b'},
    },
  };

  it('renders correctly', () =>  {
    const wrapper = Enzyme.shallow(
      <ArticleList
        {...testProps}
      />
    );

    expect(wrapper.find('ArticleContainer').length).toBe(2);
    expect(wrapper).toMatchSnapshot();

  });
});
