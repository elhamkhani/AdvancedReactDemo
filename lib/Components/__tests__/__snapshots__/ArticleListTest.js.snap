// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={Object {}}
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={Object {}}
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
  <div
    style={Object {}}
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={Object {}}
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
</div>
`;
